controllers/pipeline/pipeline_adapter.go

```diff
@@ -632,3 +613,23 @@ func (a *Adapter) isLatestSucceededPipelineRun() (bool, error) {
 	}
 	return true, nil
 }
+
+// getSucceededBuildPipelineRunsForComponent returns all  succeeded PipelineRun for the
+// associated component. In the case the List operation fails,
+// an error will be returned.
+func getSucceededBuildPipelineRunsForComponent(adapterClient client.Client, ctx context.Context, loader loader.ObjectLoader, component *applicationapiv1alpha1.Component) (*[]tektonv1beta1.PipelineRun, error) {
```

Comment:
Suggestion: dropping the client, context and loader as parameters and using `a` to reference them
```suggestion
func (a *Adapter) getSucceededBuildPipelineRunsForComponent(component *applicationapiv1alpha1.Component) (*[]tektonv1beta1.PipelineRun, error) {
```

Reply:
Agree, thanks!